### PR TITLE
Testing a new way of dealing with extension codegen (this is a POC)

### PR DESCRIPTION
**Pull request by ia3andy**
The idea is to surgically change the pom.xml and build.gradle instead of totally parse/rewrite it.

I was thinking about having a method that gets a set of lines to ignore for all future parsing (like comments for example).

The creation from scratched would be dealt with using Qute template containing the whole build files.

PROS:
- No reformatting or alteration of the buildfile (beside the actual change)
- No external library to use
- Easy to understand the system and improve
- The parsing algos are small, scoped, testable, isolated and uncoupled. They can be optimized incrementally.

CONS:
- Dealing with parent pom would be to complex (but do we really need it?)
- People hate Regex (I reduced the complexity by parsing by lines)
- We may encounter corner cases to fix along the way (or make unsupported)

**I made this PR to validate the concept, it's not optimized neither tested for every cases.**

### CODE REVIEW

File: independent-projects/tools/pom.xml [hunk 1]
```diff
@@ -54,6 +54,7 @@
     <modules>
         <module>platform-descriptor-api</module>
         <module>common</module>
+        <module>qoder</module>
```
Comment (aloubyansky) [inline independent-projects/tools/pom.xml:57]: let's probably come up with something less obscure

File: independent-projects/tools/qoder/src/test/resources/valid-pom.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            <groupId>io.quarkus</groupId>
+            <!-- My comment -->
+            <artifactId>some-ext</artifactId>
+        </dependency>
+        <dependency>
+            <!-- My comment -->
+            <groupId>org.other</groupId>
+            <artifactId>some-other-ext</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (aloubyansky) [inline independent-projects/tools/qoder/src/test/resources/valid-pom.xml:28]: As we discussed, please also add a pom with some dependencies (including extensions) commented out.

File: independent-projects/tools/pom.xml (hunk 1 above)
Comment (ia3andy) [reply independent-projects/tools/pom.xml]: changed to `quarkus-devtools-codegen`

File: independent-projects/tools/qoder/src/test/resources/valid-pom.xml (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/qoder/src/test/resources/valid-pom.xml]: I added the detection and a test for it

**Comment by ia3andy** [comment]
Closing for now as we may have a better solution..

**Pull request closed without merging**